src/permissions.rs

```diff
@@ -0,0 +1,86 @@
+extern crate atty;
+
+use flags::DenoFlags;
+
+use errors::permission_denied;
+use errors::DenoResult;
+use std::io;
+
+#[derive(Debug, Default, PartialEq)]
+pub struct DenoPermissions {
+  pub allow_write: bool,
+  pub allow_net: bool,
+  pub allow_env: bool,
+}
+
+impl DenoPermissions {
+  pub fn new(flags: &DenoFlags) -> DenoPermissions {
+    DenoPermissions {
+      allow_write: flags.allow_write,
+      allow_env: flags.allow_env,
+      allow_net: flags.allow_net,
+    }
+  }
+
+  pub fn check_write(&mut self, filename: &str) -> DenoResult<()> {
+    if self.allow_write {
+      return Ok(());
+    };
+    // TODO get location (where access occurred)
+    let r = permission_prompt(format!(
+      "🔐 Deno requests write access to \"{}\".",
+      filename
+    ));;
+    if r.is_ok() {
+      self.allow_write = true;
```

Comment:
I still think this behavior is a little weird: You're asked if it's okay for deno to write foo.txt, you say yes, and deno can write **anything** for the rest of time.

IMO for now we should not use the filename/domain arg until we have finer permissions, either:

- `"🔐 Deno requests write access.` for the rest of execution.
- `"🔐 Deno requests write access to \"{}\"."` but ask every time.

Reply:
I think it's fine for a first pass. This feature is about ergonomics and clicking "yes" a bunch of times is annoying. Maybe we can change the wording to reflect the fact that access is granted for the rest of execution.